src/AsyncEventSource.cpp

```diff
@@ -302,6 +319,8 @@ void AsyncEventSource::_addClient(AsyncEventSourceClient * client){
 
 void AsyncEventSource::_handleDisconnect(AsyncEventSourceClient * client){
   _clients.remove(client);
+  if(_disconnectcb)
+    _disconnectcb(this);
```

Comment:
Why this and not client ref ?

Reply:
Fair point.  I was just adding what I needed since I needed the parent event source.

I've now added client ref to the cb2 (but leaving the parent "this" parameter) and added the same overload for OnConnect.